### ISSUE TITLE
[Feature Request] - Make route response variables accessible from the dataset.

### ISSUE THREAD

**Issue by edtoken** [mockoon/mockoon#1103]
### Problem:
When you trying to mock some API larger than a few requests you need some reusable data, and this is covered by the dataset that could be re-used in the responses.

But, also, this API requires some customized responses of that data, and these "customizations" is something like snippets of the templates.

Example:
I have a dataset with users that could be filtered by query param and the response also includes the length of the filtered array as a param.

Basically, I could do the expected mock with instruments that Mockoon gave right now, but when I have a few methods with the same logic then I will have really a lot of boilerplate that requires a lot of time for the support.

Example: dataset filter (found on discord Mockoon page)
```
{{setVar 'addComma' false }} 
  {{#each (dataRaw 'Users' )}} 
    {{#if ( includes  this.type 'value')}}
      {{#if @addComma}},{{/if}}
      {{data 'Users' (concat '' @index)}}
      {{setVar 'addComma' true }}
    {{/if}}
  {{/each}}
```

I know that you are trying to make Mockoon a simple tool, so, I feel that a "snippets" tab is something that a little more that we probably need right now.

---
### Solution

If we allow access of the route variables inside the datasets, then i could pass from the route level 
- filter name (` {{#if ( includes  this.type @routeVariableName)}}`)
- dataset name ( `{{#each (dataRaw @routeDatasetName )}} `)

I would say that the current flexibility of the datasets with handlebars and access to other datasets 
**AND the new feature route-level variables access** could make the datasets the feature that covers whole unexpected behaviors.

- query/cookie/any param of the request based on some conditions could set specific variables
- based on the route variables we could filter/modify dataset results

I think a couple of issues that are open right now also could be covered by that small improvement

Example of the issues that could be resolved by establishing variable on route level and then use in the dataset [1098](https://github.com/mockoon/mockoon/issues/1098) [1092](https://github.com/mockoon/mockoon/issues/1092) [1047](https://github.com/mockoon/mockoon/issues/1047) [1034](https://github.com/mockoon/mockoon/issues/1034) and more

### Why not route a template?
> Just time that you need to support large API with a lot of re-usages of "snippets" that could be implemented as flexible datasets.

**Comment by edtoken** [comment]
UPD: instead of that feature you could allow just add custom handlebars helpers.
The main goal needs a little more flexibility for real usage

- How to filter data set and returns filtered (also get length of the filtered data)?
- How to re-use some snippets
- How to implement paged responses based on datasets? 
- ...

**Comment by 255kb** [comment]
📦 This is now available with [release v4.1.0](https://mockoon.com/releases/4.1.0).
